### ISSUE TITLE
Plugin Failing After IntelliJ Upgrade to 0.13.15 if jenkins is hosted Tomcat or Query Parameters are not relaxed

### ISSUE THREAD

**Issue by fiveoaksmn** [MCMicS/jenkins-control-plugin#426]
After upgrading to 2022.3, the Jenkins plugin has stopped working. Below exception is being thrown:

```
Access is allowed from event dispatch thread only

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=false
Current thread: Thread[ApplicationImpl pooled thread 63,4,main] 1748510217
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1166426085
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1042)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1025)
	at com.intellij.ui.SpeedSearchBase.isPopupActive(SpeedSearchBase.java:180)
	at com.intellij.ui.speedSearch.SpeedSearchSupply.getSupply(SpeedSearchSupply.java:63)
	at com.intellij.ui.speedSearch.SpeedSearchSupply.getSupply(SpeedSearchSupply.java:52)
	at com.intellij.ui.speedSearch.SpeedSearchUtil.applySpeedSearchHighlightingFiltered(SpeedSearchUtil.java:185)
	at com.intellij.ui.ColoredTreeCellRenderer.rendererComponentInner(ColoredTreeCellRenderer.java:124)
	at com.intellij.ui.ColoredTreeCellRenderer.getTreeCellRendererComponent(ColoredTreeCellRenderer.java:53)
	at com.intellij.ui.tree.ui.DefaultTreeUI.getRenderer(DefaultTreeUI.java:179)
	at com.intellij.ui.tree.ui.DefaultTreeUI$2.getNodeDimensions(DefaultTreeUI.java:553)
	at java.desktop/javax.swing.tree.AbstractLayoutCache.getNodeDimensions(AbstractLayoutCache.java:497)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.updatePreferredSize(VariableHeightLayoutCache.java:1344)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:723)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:629)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:4407)
	at java.desktop/javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:616)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:320)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:227)
	at org.codinjutsu.tools.jenkins.JenkinsTree.clear(JenkinsTree.java:149)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.clearView(BrowserPanel.java:313)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.setJobsUnavailable(BrowserPanel.java:308)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.handleEmptyConfiguration(BrowserPanel.java:304)
	at org.codinjutsu.tools.jenkins.logic.BrowserPanelAuthenticationHandler.loginCancelled(BrowserPanelAuthenticationHandler.java:42)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:652)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:621)
	at com.intellij.util.messages.impl.MessageBusImplKt.deliverMessage(MessageBusImpl.kt:395)
	at com.intellij.util.messages.impl.MessageBusImplKt.pumpWaiting(MessageBusImpl.kt:374)
	at com.intellij.util.messages.impl.MessageBusImplKt.access$pumpWaiting(MessageBusImpl.kt:1)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:433)
	at jdk.proxy13/jdk.proxy13.$Proxy240.loginCancelled(Unknown Source)
	at org.codinjutsu.tools.jenkins.logic.LoginService$1.run(LoginService.java:45)
	at org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask.run(JenkinsBackgroundTask.java:41)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:425)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:476)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:190)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:666)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:622)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:590)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:177)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

If I go into the preferences for the plugin and click Test Connection, it responds with: [Fail] URL is malformed.

The URL is correct however. I can copy and paste it into a browser and it works just fine.

**Comment by MCMicS** [comment]
Can you please use following: https://plugins.jetbrains.com/plugin/6110-jenkins-control/versions/eap/263516

**Comment by cfhuch** [comment]
> Can you please use following: https://plugins.jetbrains.com/plugin/6110-jenkins-control/versions/eap/263516

Works for me! Thanks! 😄

**Comment by MCMicS** [comment]
nice to hear
I will it release this version at start of new year

**Comment by fiveoaksmn** [comment]
I installed the version reference above but it is still failing for me. Not getting the exception anymore but the plugin is still saying malformed URL in the preferences and none of the tasks will load. I do have a port in my URL. Could that be making a difference?

e.g. http://127.0.0.1:8080/jenkins/

**Comment by MCMicS** [comment]
No the port is no problem. Maybe another error or redirect for your server

Can you do a curl call to test it?

**Comment by fiveoaksmn** [comment]
Curl works just fine.

**Comment by MCMicS** [comment]
Do you use any proxies?

**Comment by fiveoaksmn** [comment]
Nope. And I can access Jenkins from the web browser with no problem. The plugin was working great until I upgraded to IntelliJ 2022.3.

**Comment by MCMicS** [comment]
I will test again.
Is your server public visible or it is possible to grant access to me for test

In latest plugin version I switched the used http client implementation. Have yo tested the same plugin version with older idea version too?

**Comment by fiveoaksmn** [comment]
No, server is not public. 

I have an older copy of IntelliJ on a different computer that I can test the latest version of the plugin on. Will do in a few hours.

**Comment by fiveoaksmn** [comment]
I tested in IntelliJ 2022.2 on another computer. As soon as I upgrade the plugin to the newest version you provided the link to, it also fails to connect to Jenkins. So appears it is not related to the IntelliJ version.

**Comment by MCMicS** [comment]
Thanks
i will investigate it.
In my setups with different jenkins server it still works

I will extend the loggging to find the problem

**Comment by MCMicS** [comment]
Could you please try following buiid. and try 

And enable Debug settings if error still occurs (enable for `org.codinjutsu.tools.jenkins`). Then in ida.log should exists the reason. 
![grafik](https://user-images.githubusercontent.com/4604206/210273928-99ec21ca-127b-478d-a108-072f8bc80a75.png)

My Settings:
![grafik](https://user-images.githubusercontent.com/4604206/210274081-8a1296b2-54db-48dc-87e0-cd326133f03c.png)

Plugin: [jenkins-control-plugin-0.13.16-eap3-2022.2.zip](https://github.com/MCMicS/jenkins-control-plugin/files/10333464/jenkins-control-plugin-0.13.16-eap3-2022.2.zip)

**Comment by fiveoaksmn** [comment]
Still not working. Here is what I am seeing in the IntelliJ logs: 

2023-01-02 15:53:57,234 [ 478264] FINE - org.codinjutsu.tools.jenkins.view.action.UploadPatchToJobAction - 
java.net.MalformedURLException: no protocol: 
at java.base/java.net.URL.<init>(URL.java:674) 
at java.base/java.net.URL.<init>(URL.java:569) 
at java.base/java.net.URL.<init>(URL.java:516) 
at org.codinjutsu.tools.jenkins.logic.UrlBuilder.toUrl(UrlBuilder.java:191) 
at org.codinjutsu.tools.jenkins.logic.ConfigurationValidator.validate(ConfigurationValidator.java:32) 
at org.codinjutsu.tools.jenkins.view.ConfigurationPanel.testConnection(ConfigurationPanel.java:173) 
at org.codinjutsu.tools.jenkins.view.ConfigurationPanel.lambda$new$0(ConfigurationPanel.java:143) 
at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) 
at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) 
at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) 
at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) 
at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279) 
at java.desktop/java.awt.Component.processMouseEvent(Component.java:6648) 
at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3392) 
at java.desktop/java.awt.Component.processEvent(Component.java:6413) 
at java.desktop/java.awt.Container.processEvent(Container.java:2266) 
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022) 
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) 
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854) 
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) 
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) 
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) 
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) 
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802) 
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854) 
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781) 
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730) 
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724) 
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) 
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754) 
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752) 
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) 
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751) 
at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909) 
at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:831) 
at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:753) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437) 
at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436) 
at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105) 
at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434) 
at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813) 
at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480) 
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207) 
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128) 
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:121) 
at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191) 
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236) 
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234) 
at java.base/java.security.AccessController.doPrivileged(AccessController.java:318) 
at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234) 
at java.desktop/java.awt.Dialog.show(Dialog.java:1080) 
at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:743) 
at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:467) 
at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1676) 
at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1634) 
at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:134) 
at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:107) 
at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:99) 
at org.codinjutsu.tools.jenkins.view.action.OpenPluginSettingsAction.showSettingsFor(OpenPluginSettingsAction.java:47) 
at java.base/java.util.Optional.ifPresent(Optional.java:178) 
at org.codinjutsu.tools.jenkins.view.action.OpenPluginSettingsAction.actionPerformed(OpenPluginSettingsAction.java:43) 
at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:195) 
at com.intellij.openapi.actionSystem.impl.ActionButton.lambda$performAction$0(ActionButton.java:174) 
at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:346) 
at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:174) 
at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:462) 
at java.desktop/java.awt.Component.processEvent(Component.java:6413) 
at java.desktop/java.awt.Container.processEvent(Container.java:2266) 
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022) 
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) 
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854) 
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) 
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) 
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) 
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) 
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802) 
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854) 
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781) 
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730) 
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724) 
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) 
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754) 
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752) 
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) 
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) 
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751) 
at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909) 
at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:831) 
at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:753) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437) 
at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436) 
at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113) 
at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615) 
at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434) 
at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813) 
at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480) 
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207) 
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128) 
at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117) 
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113) 
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105) 
at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

**Comment by MCMicS** [comment]
it says `java.net.MalformedURLException: no protocol:`

Why the the protocol is missing.

The jenkins server seem not to return a valid url

Please use curl:
`curl -u user:token http://127.0.0.1:8080/jenkins/api/json?tree=nodeName,url,description,primaryView%5Bname,url%5D`

the JSOn result should looks like:
```json
{"_class":"hudson.model.Hudson","nodeName":"","description":null,"primaryView":{"_class":"hudson.model.AllView","name":"all","url":"http://127.0.0.1:8080/jenkins/"},"url":"http://127.0.0.1:8080/jenkins/"}
```

I suggest that the primary View or url is empty

**Comment by MCMicS** [comment]
I could reproduce it with following steps:
![grafik](https://user-images.githubusercontent.com/4604206/210281898-cb4ee7fc-b443-4744-acd0-31fd87189de0.png)

1. Clear Jenkins URL and click save
2. Then the JSON result contains the `null`value
3. click refresh (F5 or Ctrl/Cmd+R) and the default URL is filled in the text. Then save theses values
4. JSON should contains the correct url

**Comment by MCMicS** [comment]
I've created another version with improved url handling. I use primary view now correct as fallback

[jenkins-control-plugin-0.13.16-eap3-2022.2.zip](https://github.com/MCMicS/jenkins-control-plugin/files/10333709/jenkins-control-plugin-0.13.16-eap3-2022.2.zip)

Can you please test it?
sorry for the circumstances

**Comment by fiveoaksmn** [comment]
Still failing. I get a different error now though. When I click test connection, it says "[Fail] Cannot find 'Jenkins URL'. Please check your Jenkins location." 

URL is valid though. Only error I see in the log is following: 

2023-01-02 19:44:33,389 [ 222781] FINE - org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask - 
org.codinjutsu.tools.jenkins.exception.JenkinsPluginRuntimeException: Empty JSON data! 
at org.codinjutsu.tools.jenkins.logic.JenkinsJsonParser.checkJsonDataAndThrowExceptionIfNecessary(JenkinsJsonParser.java:528) 
at org.codinjutsu.tools.jenkins.logic.JenkinsJsonParser.createWorkspace(JenkinsJsonParser.java:62) 
at org.codinjutsu.tools.jenkins.logic.RequestManager.loadJenkinsWorkspace(RequestManager.java:107) 
at org.codinjutsu.tools.jenkins.logic.LoginService$1.run(LoginService.java:43) 
at org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask.run(JenkinsBackgroundTask.java:41) 
at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:425) 
at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114) 
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:476) 
at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252) 
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:190) 
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:591) 
at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:666) 
at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:622) 
at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:590) 
at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60) 
at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:177) 
at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)

**Comment by MCMicS** [comment]
The message `Cannot find 'Jenkins URL'. Please check your Jenkins location` is my new one if no url is found.

Can you please send me your json result for the curl command.

Have you tried to to set the server url in 'manage jenkins' again?

**Comment by fiveoaksmn** [comment]
What URL should I pass to curl to get the JSON response? If I just do the basic URL, I get a HTML page saying Authentication required.

**Comment by fiveoaksmn** [comment]
Here is the output from curl:

```
*   Trying 127.0.0.1:8080...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
* Server auth using Basic with user 'grant'
> GET /jenkins/api/json?tree=nodeName,url,description,primaryView%5Bname,url%5D HTTP/1.1
> Host: 127.0.0.1:8080
> Authorization: Basic Z3JhbnQ6MTFlNjA4N2FhNjEyMjg4MDY2OTViZWM5NmQzN2U4M2RhMw==
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200
< Cache-Control: private
< X-Content-Type-Options: nosniff
< X-Jenkins: 2.361.2
< X-Jenkins-Session: e78c5abb
< X-Frame-Options: deny
< Content-Type: application/json;charset=UTF-8
< Content-Length: 208
< Date: Wed, 04 Jan 2023 06:30:00 GMT
<
* Connection #0 to host 127.0.0.1 left intact
{"_class":"hudson.model.Hudson","nodeName":"","description":null,"primaryView":{"_class":"hudson.model.AllView","name":"all","url":"http://acorn.local:8080/jenkins/"},"url":"http://acorn.local:8080/jenkins/"}
```

I went into Jenkins and I reset the Jenkins URL. Didn't fix anything.

I'll download to an older version of the plugin tomorrow and see what happens and let you know.

**Comment by MCMicS** [comment]
Looks fine on first view
but I will look further the day

**Comment by MCMicS** [comment]
The only difference in our curl result and mine is the Header ` Cache-Control: private`.

Your Url says `http://acorn.local:8080`. Is it a local installation.
It is possible to share the jenkins configraution (zip of JENKINS_HOME)? I can also provide a space to upload for private data.

I also add additional debug logs to print out the received JSON data. Maybe I receive an Http status which I not handle correctly (like use cached values). Please set following log configuration: 
```
org.codinjutsu.tools.jenkins
#org.codinjutsu.tools.jenkins:trace
```

You will then see following in idea.log
```
o.c.t.j.s.BasicSecurityClient - Call url 'http://127.0.0.1:8080/jenkins/api/json?tree=nodeName,url,description,primaryView[name,url]' --> Status: 200, Data {"_class":"hudson.model.Hudson","nodeName":"","description":null,"primaryView":{"_class":"hudson.model.AllView","name":"all","url":"http://127.0.0.1:8080/jenkins/"},"url":null}
```

But make sure to remve the :trace entry or your log will be large.

With more logs: 
[jenkins-control-plugin-0.13.16-eap3-2022.2-signed.zip](https://github.com/MCMicS/jenkins-control-plugin/files/10346711/jenkins-control-plugin-0.13.16-eap3-2022.2-signed.zip)

**Comment by MCMicS** [comment]
Hi @fiveoaksmn 
have you got time to test it?

**Comment by fiveoaksmn** [comment]
Sorry, haven't had time yet to test. Will try to do later today.

**Comment by MCMicS** [comment]
no problem. thanks for testing

**Comment by fiveoaksmn** [comment]
Here is what is found in the IntelliJ log:

```
2023-01-06 16:14:54,273 [ 157804]  FINER - #org.codinjutsu.tools.jenkins.security.BasicSecurityClient - Call url 'http://acorn.local:8080/jenkins/api/json?tree=nodeName,url,description,primaryView[name,url]' --> Status: 400, Data <!doctype html><html lang="en"><head><title>HTTP Status 400 – Bad Request</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 400 – Bad Request</h1><hr class="line" /><p><b>Type</b> Exception Report</p><p><b>Message</b> Invalid character found in the request target [&#47;jenkins&#47;api&#47;json?tree=nodeName,url,description,primaryView[name,url] ]. The valid characters are defined in RFC 7230 and RFC 3986</p><p><b>Description</b> The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).</p><p><b>Exception</b></p><pre>java.lang.IllegalArgumentException: Invalid character found in the request target [&#47;jenkins&#47;api&#47;json?tree=nodeName,url,description,primaryView[name,url] ]. The valid characters are defined in RFC 7230 and RFC 3986
	org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:509)
	org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:513)
	org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:882)
	org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1693)
	org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	java.base&#47;java.lang.Thread.run(Thread.java:829)
```

**Comment by MCMicS** [comment]
Thanks for it
I have an idea what is the problem. It seems that your server not allow the square brackets. I will encode this tomorrow and provide you a new version for test.

have you configured your jenkins serve in special way?

I tested with simply war file and started it. But then the appi calls works.
maybe you can share your jetty or server settings

**Comment by MCMicS** [comment]
Can you please test again following build:
[Uploading jenkins-control-plugin-0.13.16-eap3-2022.2-signed.zip…]()

**Comment by fiveoaksmn** [comment]
It works! Did you just end up changing the encoding of the URL?

I am using Tomcat to host Jenkins. Nothing is configured outside the box.

I appreciate your efforts in debugging this and updating the plugin.

**Comment by MCMicS** [comment]
Yes I encoded the query parameter
I use a built-in URLUtil to encode but then square brackets was not encoded. With jetty it was no problem. In Tomcat it has to be configured with `relaxedQueryChars`

now I encode it again with another URLEncoder and it works again

I will build a release asap

Rhanks for testing

**Comment by MCMicS** [comment]
released with [0.13.16](https://plugins.jetbrains.com/plugin/6110-jenkins-control/versions/stable/277685)
